core/src/components/card/card.tsx

```diff
@@ -136,14 +150,14 @@ export class Card implements ComponentInterface, AnchorInterface, ButtonInterfac
   }
 
   render() {
-    const { shape } = this;
     const theme = getIonTheme(this);
+    const finalShape = this.getShape();
 
     return (
       <Host
         class={createColorClasses(this.color, {
           [theme]: true,
-          [`card-${shape}`]: shape !== undefined,
+          [`card-${finalShape}`]: finalShape !== undefined,
```

Comment:
Isn't a method a bit overkill? Why not just do:

```
const { shape } = this;
(...)
[`card-${finalShape}`]: theme === 'ionic' && shape !== undefined
```

Reply:
Removed this function here: https://github.com/ionic-team/ionic-framework/pull/29292/commits/3d209fc24546829f16d9e78a52e55acb9d38768a